doc/build-windows-mingw.md

```diff
@@ -0,0 +1,240 @@
+1. Prepare your build system.
+
+1.1 Install msys shell:
+http://sourceforge.net/projects/mingw/files/Installer/mingw-get-setup.exe/download
+From MinGW installation manager -> All packages -> MSYS
+mark the following for installation:
+
+msys-base-bin
+msys-autoconf-bin
+msys-automake-bin
+msys-libtool-bin
+
+then click on Installation -> Apply changes
+
+Make sure no mingw packages are checked for installation or present from a previous install. Only the above msys packages should be installed. Also make sure that msys-gcc and msys-w32api packages are not installed.
+
+1.2 Install MinGW-builds project toolchain:
+Download http://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win32/Personal%20Builds/mingw-builds/4.9.2/threads-posix/dwarf/i686-4.9.2-release-posix-dwarf-rt_v3-rev1.7z/download
+and unpack it to C:\
+
+1.3. Ensure that mingw-builds bin folder is set in your PATH environment variable. On Windows 7 your path should look something like:
+
+Code:
+
+C:\mingw32\bin;%SystemRoot%\system32;%SystemRoot%;%SystemRoot%\System32\Wbem;%SYSTEMROOT%\System32\WindowsPowerShell\v1.0\
+
+
+1.4 Additional checks:
+C:\MinGW\bin should contain nothing but mingw-get.exe.
+Your gcc -v output should be:
+
+Code:
+
+$ gcc -v
+Using built-in specs.
+COLLECT_GCC=c:\mingw32\bin\gcc.exe
```

Comment:
Minor difference if you run in a Windows CMD window or the msys shell is that the windows CMD window only shows gcc while the shell shows the full path to the exe.